neuralpp/symbolic/basic_expression.py

```diff
@@ -53,10 +53,9 @@ def __init__(self, atom: Any, expression_type: Optional[ExpressionType] = None):
         if expression_type is None and not isinstance(atom, ExpressionType):
             # try to infer type for atom
             if isinstance(atom, Callable):
-                internal_type = infer_python_callable_type(atom)
+                expression_type = infer_python_callable_type(atom)
             else:
-                internal_type = type(atom)
-            expression_type = BasicConstant(internal_type, None)
+                expression_type = type(atom)
```

Comment:
It was a bug, we no longer need to wrap the type with `Constant()`, doing so causes errors.